### PR TITLE
add docs around pipelines obj store configurations

### PR DESCRIPTION
**Pull request by HumairAK**
This documents the features introduced via: 

* https://github.com/kubeflow/pipelines/pull/10625
* https://github.com/kubeflow/pipelines/pull/10787

Quick link to netlify page preview: https://deploy-preview-3833--competent-brattain-de2d6d.netlify.app/docs/components/pipelines/operator-guides/configure-object-store/

### CODE REVIEW

**Comment by HumairAK** [comment]
/hold

This should not be merged until KFP 2.3 released (no timelines on this yet)

**Comment by HumairAK** [comment]
cc @thesuperzapper

**Comment by gregsheremeta** [comment]
looks good to me! I'll wait to tag it so that others have a chance to provide feedback.

sidebar: the different yamls for all the different ways to setup the various object stores gets confusing really fast, and feels like it would be prone to getting misconfigured easily. I wonder if there could be some way to abstract all of that cruft behind some nice "data connection" facade...

File: content/en/docs/components/pipelines/concepts/pipeline-root.md [hunk 1]
```diff
@@ -45,13 +44,9 @@ If you want to specify the `pipeline root` to S3, please choose one of the follo
 ## How to configure pipeline root
 
 #### Via ConfigMaps
+The default Pipeline root at the Kubeflow pipeline deployment level by configuring the KFP Launcher configmap.
```
Comment (anishasthana) [inline content/en/docs/components/pipelines/concepts/pipeline-root.md:47]: `The default Pipeline root at the Kubeflow pipeline deployment level by configuring the KFP Launcher configmap.`

What are you trying to say here exactly? Should this say 
`The default Pipeline root at the Kubeflow pipeline deployment level can be changed by configuring the KFP Launcher configmap. ` or something similar?

File: content/en/docs/components/pipelines/operator-guides/configure-object-store.md [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+| Minio with Static Credentials                | Yes       |
+| AWS S3 with Static Credentials               | Yes       |
+| AWS S3 with IRSA                             | Yes       |
+| S3-Compliant Storage with Static Credentials | Yes       |
+| Google Cloud Storage with Static Credentials | No        |        
+| Google Cloud Storage with App Credentials    | No        |  
+
+### API Server Object Store Configuration
+
+To configure the object store used by the KFP API Server, the configuration depends on whether you are using static credentials, or AWS S3 with IRSA.
```
Comment (rimolive) [inline content/en/docs/components/pipelines/operator-guides/configure-object-store.md:52]: I think IRSA is not the best way to mention. Refer to the full name, then add the acronym in parenthesis.

Worth mentioning that a reference link would be also good.

File: content/en/docs/components/pipelines/operator-guides/configure-object-store.md [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+                key: "some-key-1"
+                name: "secret-name"
+          - name: OBJECTSTORECONFIG_SECRETACCESSKEY
+            valueFrom:
+              secretKeyRef:
+                key: "some-key-2"
+                name: "secret-name"
+```
+
+**AWS IRSA (IAM Roles for Service Accounts)**
```
Comment (rimolive) [inline content/en/docs/components/pipelines/operator-guides/configure-object-store.md:98]: Take a look at my previous comment. The full name should be referenced the first time you mention it

File: content/en/docs/components/pipelines/concepts/pipeline-root.md (hunk 1 above)
Comment (HumairAK) [reply content/en/docs/components/pipelines/concepts/pipeline-root.md]: yes, thanks, fixed

File: content/en/docs/components/pipelines/operator-guides/configure-object-store.md (hunk 2 above)
Comment (HumairAK) [reply content/en/docs/components/pipelines/operator-guides/configure-object-store.md]: fixed

File: content/en/docs/components/pipelines/operator-guides/configure-object-store.md [hunk 4]
```diff
@@ -0,0 +1,404 @@
++++  
+title = "Object Store Configuration"  
+weight = 3  
++++
+
+{{% kfp-v2-keywords %}}
+
+In Kubeflow Pipelines (KFP) are two components that utilize Object store:
```
Comment (VaniHaripriya) [inline content/en/docs/components/pipelines/operator-guides/configure-object-store.md:8]:
```suggestion
In Kubeflow Pipelines (KFP), there are two components that utilize Object store:
```

**Comment by franciscojavierarceo** [comment]
lgtm

**Pull request merged**